### PR TITLE
docs: Add BackboneLambdaFinetuningCallback

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
The docs of `BackboneLambdaFinetuningCallback` and `BaseFinetuningCallback` aren't available right now, which was probably missed in #5377.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/callbacks.rst [hunk 1]
```diff
@@ -94,6 +94,8 @@ Lightning has a few built-in callbacks.
     :nosignatures:
     :template: classtemplate.rst
 
+    BaseFinetuningCallback
+    BackboneLambdaFinetuningCallback
```
Comment (awaelchli) [inline docs/source/callbacks.rst:98]: this one comes first alphabetically

**Review by awaelchli — APPROVED** [review]

File: docs/source/callbacks.rst (hunk 1 above)
Comment (akihironitta) [inline docs/source/callbacks.rst:98]:
```suggestion
    BackboneLambdaFinetuningCallback
    BaseFinetuningCallback
```

File: docs/source/callbacks.rst [hunk 2]
```diff
@@ -94,6 +94,8 @@ Lightning has a few built-in callbacks.
     :nosignatures:
     :template: classtemplate.rst
 
+    BackboneLambdaFinetuningCallback
+    BaseFinetuningCallback
```
Comment (SkafteNicki) [inline docs/source/callbacks.rst:98]: Should `BaseFinetuningCallback` be included in the documentation?
The documentation on this class very sparse. For example it should atleast state in class docstring that the user needs to implement the `finetunning_function` and `freeze_before_training` method to get a working subclass.

Comment (akihironitta) [reply docs/source/callbacks.rst:98]: As `BaseFinetuningCallback` is currently part of public API, I'd include the class in the docs, or we could make the class private if it's not supposed to be exposed to users. What do you think?

Comment (SkafteNicki) [reply docs/source/callbacks.rst:98]: If it is part of the public API then we should probably include the class in the docs (else we need to deprecate it 2 releases from now). IMO we should probably improve the documentation of this class.

Comment (tchaton) [reply docs/source/callbacks.rst:98]: Yes, I was planning to spend some time on the doc and provide a better name to `BackboneLambdaFinetuningCallback ` which is too long.
Thanks @akihironitta to worked on this.

**Review by tchaton — APPROVED** [review]

File: docs/source/callbacks.rst (hunk 2 above)
Comment (SkafteNicki) [reply docs/source/callbacks.rst:98]: @tchaton sounds good to me :]

**Review by SkafteNicki — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Comment by tchaton** [comment]
Hey @akihironitta @awaelchli @carmocca ,

After reflexion, I realise the Callback names aren't great !

Any proposition for better names ?

Best,
T.C

**Comment by carmocca** [comment]
FinetuneCallback or FinetuningCallback are the simplest
